proto/proto/responses.proto

```diff
@@ -60,3 +60,28 @@ message SyncStateResponse {
     // a list of nullifiers created between `block_ref` and `block_header.block_num`
     repeated NullifierUpdate nullifiers = 7;
 }
+
+message AccountInputRecord {
+    account_id.AccountId account_id = 1;
+    digest.Digest account_hash = 2;
+    merkle.MerklePath proof = 3;
+}
```

Comment:
My original proposal here was to send a `PartialMerkleTree` rather than individual Merkle proofs - but now I'm not so sure. The lay out pros and cons:

- The main benefit of sending a `PartialMerkleTree` is that we save on bandwidth as Merkle paths are likely to have many duplicate nodes at the top.
- The main benefit of sending just a list of Merkle paths is simplicity (and given our current implementation of the `PartialMerkleTree`, this would also be more computationally efficient). We can also optimize each individual path quite a bit by compacting them (i.e., using a simple bitmap to encode nodes which are roots of empty subtrees).

The bandwidth is unlikely to become an issue until we have blocks with well over 1000 transactions (even then, this endpoint will probably not return more than a few MB of data). So, my inclination is to leave it as is and create an issue to improve this after the first testnet.

If we do the above, this leads to a question of what to do with the nullifier portion of the response. There, due to the complexity of the TSMT, we need to return some extra data (e.g., a map from a leaf of TSMT to the key-value pairs it commits to). Basically, the `NullifierInputRecord` below would look something like this:

```proto
message NullifierInputRecord {
    digest.Digest leaf = 1;
    repeat KeyValuePair = 2;
    merkle.MerklePath proof = 3;
    uint64 index = 4;
}
```
In the above, I'm assuming that `KeyValuePair` is just a pair of digests. It should be possible to construct the above from `TieredSmtInsertWitness` (being worked on in https://github.com/0xPolygonMiden/crypto/pull/211) - but maybe there is a better way as well.

cc @plafer.

Reply:
> The main benefit of sending a PartialMerkleTree is that we save on bandwidth as Merkle paths are likely to have many duplicate nodes at the top.

The account id is uniformly distributed, so with every bit the likelyhood of sharing a node decreases by half, we should end with something like `lg(n)` shared nodes on the top, which I guess will amount to 3/4 nodes (is there are reason for clients to have more than a handful of accounts?)

Reply:
This endpoint will be used by the block producer rather than the client. At high TPS, the number of requested accounts could be several thousand (or even more) - so, we may end up with 10 - 12 nodes at the top duplicated. But not an issue until we get to these high TPS.

Reply:
Yep, so the total saving would be ~320KB (320 bytes per path x 1K paths) but even that is probably not too big of a deal (we'll probably get much more benefit from compressed Merkle paths).